packages/vision/src/index.js

```diff
@@ -20,2097 +20,45 @@
- * var image = new Buffer(myBase64ImageRepresentation, 'base64');
- *
- * vision.detect(image, types, function(err, detection, apiResponse) {});
- *
- * //-
- * // Supply multiple images for feature detection.
- * //-
- * var images = [
- *   'image.jpg',
- *   'image-two.jpg'
- * ];
- *
- * var types = [
- *   'face',
- *   'label'
- * ];
- *
- * vision.detect(images, types, function(err, detections, apiResponse) {
- *   // detections = [
- *   //   // Detections for image.jpg:
- *   //   {
- *   //     faces: [...],
- *   //     labels: [...]
- *   //   },
- *   //
- *   //   // Detections for image-two.jpg:
- *   //   {
- *   //     faces: [...],
- *   //     labels: [...]
- *   //   }
- *   // ]
- * });
- *
- * //-
- * // It's possible for part of your request to be completed successfully, while
- * // a single feature request was not successful.
- * //-
- * vision.detect('malformed-image.jpg', types, function(err, detections) {
- *   if (err) {
- *     // An API error or partial failure occurred.
- *
- *     if (err.name === 'PartialFailureError') {
- *       // err.errors = [
- *       //   {
- *       //     image: 'malformed-image.jpg',
- *       //     errors: [
- *       //       {
- *       //         code: 400,
- *       //         message: 'Bad image data',
- *       //         type: 'faces'
- *       //       },
- *       //       {
- *       //         code: 400,
- *       //         message: 'Bad image data',
- *       //         type: 'labels'
- *       //       }
- *       //     ]
- *       //   }
- *       // ]
- *     }
- *   }
- *
- *   // `detections` will still be populated with all of the results that could
- *   // be annotated.
- * });
- *
- * //-
- * // If the callback is omitted, we'll return a Promise.
- * //-
- * vision.detect('image.jpg', types).then(function(data) {
- *   var detections = data[0];
- *   var apiResponse = data[1];
- * });
+ * @param {Object=} opts - The optional parameters.
+ * @param {String=} opts.servicePath
+ *   The domain name of the API remote host.
+ * @param {number=} opts.port
+ *   The port on which to connect to the remote host.
+ * @param {grpc.ClientCredentials=} opts.sslCreds
+ *   A ClientCredentials for use with an SSL-enabled channel.
+ * @param {Object=} opts.clientConfig
+ *   The customized config to build the call settings. See
+ *   {@link gax.constructSettings} for the format.
  */
-Vision.prototype.detect = function(images, options, callback) {
-  var self = this;
-  var isSingleImage = !is.array(images) || images.length === 1;
+function visionV1(opts) {
+  // Define the header options.
+  opts = opts || {};
+  opts.libName = 'gccl';
+  opts.libVersion = VERSION;
 
-  if (!is.object(options)) {
-    options = {
-      types: options
-    };
+  // Create the image annotator client with the provided options.
+  var client = gapic.v1(opts).imageAnnotatorClient(opts);
+  if (is.undefined(client.annotateImage)) {
+    Object.assign(client.constructor.prototype, helpers('v1'));
```

Comment:
Like the feature methods defined inside the helpers.js file, this way of programmatically defining methods is a problem for the docs. Off hand, I'm not exactly sure how to solve for this.